### PR TITLE
feed2imap-go: init at 1.6.0

### PR DESCRIPTION
**Pull request by nomeata**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/feedreaders/feed2imap-go/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-Ywo+ll6Ysy7H3bfzTBsT8U7M9XVOBlfd0v7RMgKoitY=";
+  };
+
+  vendorHash = "sha256-dtui0eChZXTrvprX3py/xCGmbHUYhk7nCHQ9pwBMHRY=";
+
+  postInstall = ''
+    # Move the print-cache tool out of the PATH, seems to be an internal development tool
+    mkdir $out/libexec
+    mv $out/bin/print-cache $out/libexec
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/feedreaders/feed2imap-go/default.nix:19]: Is this even used or does it work in this location without extra patches? if not we can just delete it

File: pkgs/applications/networking/feedreaders/feed2imap-go/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Uploads rss feeds as e-mails onto an IMAP server";
+    homepage = "https://github.com/Necoro/feed2imap-go";
+    license = licenses.gpl2;
+    platforms = platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/feedreaders/feed2imap-go/default.nix:26]: this seems like it is pure go, so it should work on darwin, too and we can just delete this line

File: pkgs/applications/networking/feedreaders/feed2imap-go/default.nix (hunk 1 above)
Comment (nomeata) [reply pkgs/applications/networking/feedreaders/feed2imap-go/default.nix:19]: Good question. I assumed it was a debugging tool; not worth having in the PATH, but good to have around. I’m asking upstream at https://github.com/Necoro/feed2imap-go/issues/94

File: pkgs/applications/networking/feedreaders/feed2imap-go/default.nix (hunk 2 above)
Comment (nomeata) [reply pkgs/applications/networking/feedreaders/feed2imap-go/default.nix:26]:
```suggestion
```

File: pkgs/applications/networking/feedreaders/feed2imap-go/default.nix (hunk 1 above)
Comment (nomeata) [reply pkgs/applications/networking/feedreaders/feed2imap-go/default.nix]: Resolved, the binary should be deleted. Updated and squashed this branch.

**Pull request merged**